### ISSUE TITLE
Pod Created with PriorityClass fails to initialize

### ISSUE THREAD

**Issue by abhinavsinha001** [kubernetes/kubernetes#55583]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

**What happened**:
After enabling PodPriority feature-gates in API Server and Scheduler as per the document here https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/
Also enabled Priority in admission-control of API Server.
I was able to Create PriorityClass but got following error while creating pods with or without priority class after the changes:
```
Error validating pod nginx_default(8e139687-c655-11e7-9114-246e967a4158) from api, ignoring: [spec.priorityClassName: Forbidden: Pod priority is disabled by feature-gate, spec.priority: Forbidden: Pod priority is disabled by feature-gate]
```
This happens for all deployments and cluster becomes unusable. Am I missing any other configuration that needs to be enabled?

**What you expected to happen**:
I expected the deployments/pods  to initialize successfully with default / specified priority.
**How to reproduce it (as minimally and precisely as possible)**:
Enable following flags in cluster:
API Server:
```
--feature-gates=PodPriority=true
--runtime-config=scheduling.k8s.io/v1alpha1=true
--admission-control=...,Priority
```
Scheduler:
```
--feature-gates=PodPriority=true
```

Create High Priority Class:
```yaml
apiVersion: scheduling.k8s.io/v1alpha1
kind: PriorityClass
metadata:
  name: high-priority
value: 100000
globalDefault: false
description: "This priority class for high priority Applications"
```
Create a Pod using the priority class name:
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: nginx
  labels:
    env: test
spec:
  containers:
  - name: nginx
    image: nginx
    imagePullPolicy: IfNotPresent
  priorityClassName: high-priority
```

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.3", GitCommit:"f0efb3cb883751c5ffdbe6d515f3cb4fbe7b7acd", GitTreeState:"clean", BuildDate:"2017-11-08T18:39:33Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.3", GitCommit:"f0efb3cb883751c5ffdbe6d515f3cb4fbe7b7acd", GitTreeState:"clean", BuildDate:"2017-11-08T18:27:48Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration: Bare Metal amd64
- OS (e.g. from /etc/os-release): CentOs 7.4
- Kernel (e.g. `uname -a`):  4.10.10-1.el7.elrepo.x86_64
- Install tools: kubeadm 
- Others: Docker 17.10

**Comment by tengqm** [comment]
have you tried `--feature-gates=PodSecurity` ? feature gates takes a comma delimited string, you don't have to add `=true`, maybe that is causing the failure.

**Comment by dixudx** [comment]
> feature gates takes a comma delimited string, you don't have to add =true, maybe that is causing the failure.

The syntax is right.

**Comment by abhinavsinha001** [comment]
Thanks @dixudx , @tengqm  will correct the changes and verify.

**Comment by tengqm** [comment]
yep, I was wrong. It should have the `=true` part.

**Comment by abhinavsinha001** [comment]
> @abhinavsinha001 Please check PodSecurity=true has already been enabled in the feature-gates in kubelet node as well.

@dixudx can you confirm if PodSecurity is actually a feature-gate I am unable to find any documentation on this as well as I see error when I add these flags to my kubeadm.yaml which I am using for initialization. Any steps on enabling this flag will be helpful:

This is the yaml I tried:
```yaml
kind: MasterConfiguration
apiVersion: kubeadm.k8s.io/v1alpha1
networking:
  podSubnet: "10.244.0.0/16"
controllerManagerExtraArgs:
  horizontal-pod-autoscaler-use-rest-clients: "true"
apiServerExtraArgs:
  runtime-config: "api/all=true"
  feature-gates: "PodPriority=true,PodSecurity=true"
schedulerExtraArgs:
  feature-gates: "PodPriority=true,PodSecurity=true"
kubernetesVersion: "1.8.3"
```
And got following error:
```
[preflight] WARNING: kube-apiserver: failed to parse extra argument --feature-gates=PodSecurity=true
[preflight] WARNING: kube-scheduler: failed to parse extra argument --feature-gates=PodSecurity=true
```

**Comment by dixudx** [comment]
@abhinavsinha001 It is `kubelet`, not `kubeadm`.

**Comment by abhinavsinha001** [comment]
Here is where I added the flags:
```
Environment="KUBELET_EXTRA_ARGS=--fail-swap-on=false --feature-gates=PodPriority=true,PodSecurity=true"
ExecStart=
ExecStart=/usr/bin/kubelet $KUBELET_KUBECONFIG_ARGS $KUBELET_SYSTEM_PODS_ARGS $KUBELET_NETWORK_ARGS $KUBELET_DNS_ARGS $KUBELET_AUTHZ_ARGS $KUBELET_CADVISOR_ARGS $KUBELET_CGROUP_ARGS $KUBELET_CERTIFICATE_ARGS $KUBELET_EXTRA_ARGS
```

I get following error on restarting kubelet:
```
 Process: 32209 ExecStart=/usr/bin/kubelet $KUBELET_KUBECONFIG_ARGS $KUBELET_SYSTEM_PODS_ARGS $KUBELET_NETWORK_ARGS $KUBELET_DNS_ARGS $KUBELET_AUTHZ_ARGS $KUBELET_CADVISOR_ARGS $KUBELET_CGROUP_ARGS $KUBELET_CERTIFICATE_ARGS $KUBELET_EXTRA_ARGS (code=exited, status=1/FAILURE)
 Main PID: 32209 (code=exited, status=1/FAILURE)
```

If I remove PodSecurity it starts correctly

**Comment by abhinavsinha001** [comment]
My bad I only enabled the feature-gate on master node. I was successfully able to create the pods with priority after setting --feature-gates=PodPriority=true for kubelet on all Nodes.

I believe the documentation should also include that the feature-gate  needs to be enabled for kubelet as well. Added a pull with documentation update https://github.com/kubernetes/website/pull/6334

Thanks @dixudx for helping out here.
